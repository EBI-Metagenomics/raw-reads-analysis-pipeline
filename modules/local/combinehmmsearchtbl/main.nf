process COMBINEHMMSEARCHTBL {
    tag "${meta.id}"
    label 'process_single'

    input:
    tuple val(meta), path(input_files, stageAs: "inputs/?/*")

    output:
    tuple val(meta), path("${output_name}"), emit: concatenated_result

    script:
    def args = task.ext.args ?: ''
    output_name = (input_files instanceof List ? input_files[0] : input_files).name.tokenize('/')[-1]

    """
    export inputs=\$(find -L inputs -name "*tbl.gz")
    export first_file=\$(echo \$inputs | head -n 1)
    gunzip -c \${first_file} | grep '#' > header.txt || true
    cat \$inputs | gunzip | grep -v '#' > without_header.txt || true
    cat header.txt without_header.txt | gzip > ${output_name} || true
    """

    stub:
    def args = task.ext.args ?: ''
    output_name = input_files[0].name.tokenize('/')[-1]

    """
    touch ${output_name}
    """
}

