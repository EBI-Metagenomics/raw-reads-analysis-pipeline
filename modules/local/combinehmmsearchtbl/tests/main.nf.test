nextflow_process {

    name "Test Process COMBINEHMMSEARCHTBL"
    script "../main.nf"
    process "COMBINEHMMSEARCHTBL"

    tag "modules"
    tag "modules_"
    tag "combinehmmsearchtbl"

    test("one domtbl should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		    ]
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("two domtbls should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		    ]
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("10 domtbls should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    [
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		        file("${projectDir}/tests/fixtures/combinehmmsearchtbl/ERR10889197_mini.domtbl", checkIfExists: true),
		    ]
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
