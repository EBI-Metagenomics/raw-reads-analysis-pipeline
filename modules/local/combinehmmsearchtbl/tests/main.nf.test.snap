{
    "10 domtbls should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889197_mini.domtbl:md5,2e16ac4e90acd642a8ff8aef15ac8611"
                    ]
                ],
                "concatenated_result": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889197_mini.domtbl:md5,2e16ac4e90acd642a8ff8aef15ac8611"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T14:06:32.631716346"
    },
    "one domtbl should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889197_mini.domtbl:md5,a433f5f0ab57cc5837c9cbdf0ca9ffae"
                    ]
                ],
                "concatenated_result": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889197_mini.domtbl:md5,a433f5f0ab57cc5837c9cbdf0ca9ffae"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T14:06:03.223315833"
    },
    "two domtbls should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889197_mini.domtbl:md5,1f23a563727db2fd916808549355898a"
                    ]
                ],
                "concatenated_result": [
                    [
                        {
                            "id": "test"
                        },
                        "ERR10889197_mini.domtbl:md5,1f23a563727db2fd916808549355898a"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T14:06:18.012631179"
    }
}